src/ddspdrum/module.py

```diff
@@ -239,6 +239,38 @@ def make_argument(self, control_as_frequency: np.array):
         up_sampled = self.control_to_sample_rate(control_as_frequency)
         return np.cumsum(2 * np.pi * up_sampled / SAMPLE_RATE)
 
+    def oscillator(self, argument):
+        return np.cos(argument)
+
+
+class SquareSaw(VCO):
+    """ VCO that can be either a square or a sawtooth waveshape. Tweak with the shape parameter.
+
+        With apologies to:
+
+        Lazzarini, Victor, and Joseph Timoney. "New perspectives on distortion synthesis for 
+            virtual analog oscillators." Computer Music Journal 34, no. 1 (2010): 28-40.
+    """
+
+    def __init__(
+        self, 
+        shape: float = 0,
+        midi_f0: float = 10,
+        mod_depth: float = 50, 
+        phase: float = 0    
+    ):
+        super().__init__(midi_f0, mod_depth, phase)
+        assert 0 <= shape <= 1 
+        self.shape = shape
+
+    def oscillator(self, argument):
+        k = self.get_k()
+        square = np.tanh(np.pi * k * np.sin(argument)/2)
```

Comment:
I'd be very curious to see this plotted in example.py but not essential by any means

Reply:
The square fn

Reply:
it's just a clipped sin, basically. kindof a clever paper. but it's guts stuff, not sure if it needs to be upfront.